src/hotspot/share/opto/idealGraphPrinter.cpp

```diff
@@ -862,24 +904,88 @@ void IdealGraphPrinter::walk_nodes(Node* start, bool edges) {
   }
 }
 
-void IdealGraphPrinter::print_graph(const char* name) {
+static bool skip_frame(const char* name) {
+  return strstr(name, "IdealGraphPrinter") != nullptr ||
+         strstr(name, "Compile::print_method") != nullptr ||
+         strstr(name, "Compile::igv_print_graph") != nullptr ||
+         strstr(name, "PrintBFS") != nullptr ||
+         strstr(name, "Node::dump_bfs") != nullptr;
+}
+
+static bool stop_frame_walk(const char* name) {
+  return strstr(name, "C2Compiler::compile_method") != nullptr;
+}
+
+void IdealGraphPrinter::print_stack(frame fr, outputStream* graph_name) {
+  char buf[O_BUFLEN];
+  Thread* _current = Thread::current_or_null();
```

Comment:
Is this a local variable? If so, I would drop the underscore. It generally suggests that it is a field, right?
It would also not hurt to call it `current_thread` for clarity.

Reply:
Thanks, the underscore was just a leftover from copying some code from `NativeStackPrinter::print_stack_from_frame()`. I just inlined `Thread::current_or_null()` into its only use, which I think is even clearer (commit `13604b75`).